lib/matplotlib/axes/_axes.py

```diff
@@ -5164,20 +5187,32 @@ def fill_between(self, x, y1, y2=0, where=None, interpolate=False,
             where = True
         else:
             where = np.asarray(where, dtype=bool)
-            if where.size != x.size:
+            if where.size != x.size and step != 'between':
                 cbook.warn_deprecated(
                     "3.2",
                     message="The parameter where must have the same size as x "
                             "in fill_between(). This will become an error in "
                             "future versions of Matplotlib.")
-        where = where & ~functools.reduce(np.logical_or,
-                                          map(np.ma.getmask, [x, y1, y2]))
 
-        x, y1, y2 = np.broadcast_arrays(np.atleast_1d(x), y1, y2)
+        if step == 'between':
+            pad_size = x.size - 1
+        else:
```

Comment:
This is a bit confusing to me.  If we are using `between` we don't need to do any broadcasting, right?  Or if this is supposed to support `x` that is not a vector, then `x.size-1` must not be correct?  Maybe I'd need to play with this, but I'm not following what you are doing here.

Reply:
It is used when setting y2 as scalar.